runtime/criusupport/criusupport.cpp

```diff
@@ -491,6 +491,19 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 			/* throw the pending exception */
 			goto wakeJavaThreads;
 		}
+		/* At this point, Java threads are all finished, and JVM is considered paused before taking the checkpoint. */
+		vm->checkpointState.checkpointRestoreTimeDelta = 0;
+		vm->portLibrary->nanoTimeMonotonicClockDelta = 0;
+		checkpointNanoTimeMonotonic = j9time_nano_time();
+		checkpointNanoUTCTime = j9time_current_time_nanos(&success);
+		if (0 == success) {
+			systemReturnCode = errno;
+			currentExceptionClass = vm->criuJVMCheckpointExceptionClass;
+			nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_JCL_CRIU_J9_CURRENT_TIME_NANOS_FAILURE, NULL);
+			goto wakeJavaThreadsWithExclusiveVMAccess;
+		}
+		Trc_CRIU_before_checkpoint(currentThread, checkpointNanoTimeMonotonic, checkpointNanoUTCTime);
```

Comment:
this tracepoint should be printed as later as possible before the criu_dump() is called

Reply:
Could you elaborate a bit about the need to move this tracepoint assuming the calculation of `checkpointNanoTimeMonotonic`/`checkpointNanoUTCTime` is unchanged?

Reply:
Moved `Trc_CRIU_before_checkpoint()`/`Trc_CRIU_after_checkpoint()` closer to `criu_dump`, and added `Trc_CRIU_checkpoint_nano_times()`/`Trc_CRIU_restore_nano_times()` for time compensation data.

@tajila this is ready for another look.